src/psyclone/tests/psyir/nodes/node_test.py

```diff
@@ -269,6 +269,76 @@ def test_node_view_error():
             in str(error.value))
 
 
+def test_compute_cached_abs_positions(fortran_reader):
+    '''
+    Test that the Node compute_cached_abs_positions work correctly, and that
+    the invalidation and getter/setters work correctly.
+    '''
+    code = """subroutine some_routine()
+    integer :: a, b, c, d
+
+    a = 2
+    b = 3 + 4
+    c = b + a
+    if (c > 8) then
+        d = c * (b + a)
+    end if
+    end subroutine some_routine
+    """
+    psyir = fortran_reader.psyir_from_source(code)
+    # Check the caches are initially unset.
+    for child in psyir.walk(Node):
+        assert child.cached_abs_position is None
+    psyir.children[0].children[0].compute_cached_abs_positions()
+    for child in psyir.walk(Node):
+        # Compare to original abs_position implementation
+        if child is child.root:
+            assert child.cached_abs_position == child.START_POSITION
+        else:
+            found, position = child._find_position(child.root.children,
+                                                   child.START_POSITION)
+            assert found
+            assert position == child.cached_abs_position
+            # Also the abs_position should use the cached value too.
+            assert child.abs_position == child.cached_abs_position
+    # Force update the setter of the root to None and check the cache
+    # is invalidated
+    psyir.cached_abs_position = None
+    for child in psyir.walk(Node):
+        assert child.cached_abs_position is None
+    # Recompute the cache.
+    psyir.children[0].children[0].compute_cached_abs_positions()
```

Comment:
I think we can remove this (is already implicitly tested below)

Reply:
Yeah, done.